### PR TITLE
[Doc] added the ILM pattern note

### PR DESCRIPTION
**Pull request by andsel**
ILM pattern number part is 0-padded 6 digits wide string

Thanks for contributing to Logstash! If you haven't already signed our CLA, here's a handy link: https://www.elastic.co/contributor-agreement/

### CODE REVIEW

File: docs/index.asciidoc [hunk 1]
```diff
@@ -523,6 +523,8 @@ NOTE: Updating the pattern will require the index template to be rewritten
 
 NOTE: The pattern must finish with a dash and a number that will be automatically incremented when indices rollover.
 
+NOTE: The patter number is a string zero-padded 6 digits wide, see {ref}/indices-rollover-index.html#rollover-index-api-path-params[Rollover path parameters API docs] for details
```
Comment (pujithkurunji) [inline docs/index.asciidoc:526]: *pattern

Comment (andsel) [reply docs/index.asciidoc]: Thanks @pujithkurunji to point it out, fixed :-)

File: docs/index.asciidoc [hunk 2]
```diff
@@ -523,6 +523,8 @@ NOTE: Updating the pattern will require the index template to be rewritten
 
 NOTE: The pattern must finish with a dash and a number that will be automatically incremented when indices rollover.
 
+NOTE: The pattern number is a string zero-padded 6 digits wide, see {ref}/indices-rollover-index.html#rollover-index-api-path-params[Rollover path parameters API docs] for details
```
Comment (karenzone) [inline docs/index.asciidoc:526]:
```suggestion
NOTE: The pattern is a 6-digit string padded by zeros, regardless of prior index name. Example: 000001.
See {ref}/indices-rollover-index.html#rollover-index-api-path-params[Rollover path parameters API docs] for details.
```

**Review by karenzone — COMMENTED** [review]
Thanks for adding this important detail, @andsel.  I left comments for a possible rewording in line. 
Also this piece of information seems like an important part of the description and belong there instead of in a note.

**Review by karenzone — APPROVED** [review]
LGTM. Thanks for adding this info

**Pull request merged**